staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -120,6 +120,37 @@ type Config struct {
 	// Version string
 }
 
+var _ fmt.Stringer = new(Config)
+var _ fmt.GoStringer = new(Config)
+
+// GoString implements fmt.GoStringer and sanitizes sensitive fields of Config
+// to prevent accidental leaking via logs.
+func (c *Config) GoString() string {
+	return c.String()
+}
+
+// String implements fmt.Stringer and sanitizes sensitive fields of Config to
+// prevent accidental leaking via logs.
+func (c *Config) String() string {
+	if c == nil {
+		return "<nil>"
+	}
+	cc := CopyConfig(c)
+	// Explicitly mark non-empty credential fields as redacted.
+	if cc.Password != "" {
+		cc.Password = "--- REDACTED ---"
+	}
+	if cc.BearerToken != "" {
+		cc.BearerToken = "--- REDACTED ---"
+	}
+
+	// Note: below format string skips Impersonate, AuthProvider,
+	// AuthConfigPersister and ExecProvider fields as they might carry
+	// sensitive data.
```

Comment:
can we provide some indication the AuthProvider/AuthConfigPersister/ExecProvider fields were present, but omitted (if they were present)?

Reply:
can we redact just the following instead:
* config values of AuthProvider
* args and envvar values of ExecProvider